closure/goog/array/array.js

```diff
@@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
  *     for every element. This function takes 3 arguments (the element, the
  *     index and the array) and should return a boolean.
  * @param {S=} opt_obj An optional "this" context for the function.
- * @return {T|null} The first array element that passes the test, or null if no
- *     element is found.
+ * @return {T|string|null} The first array element that passes the test, or
```

Comment:
I would rather prohibit string as an input than force string to be included
in the return type.
On Feb 12, 2016 11:35 PM, "Michael Zhou" <notifications@github.com> wrote:

> In closure/goog/array/array.js
> <https://github.com/google/closure-library/pull/655#discussion_r52824745>:
>
> > @@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
> >   *     for every element. This function takes 3 arguments (the element, the
> >   *     index and the array) and should return a boolean.
> >   * @param {S=} opt_obj An optional "this" context for the function.
> > - * @return {T|null} The first array element that passes the test, or null if no
> > - *     element is found.
> > + * @return {T|string|null} The first array element that passes the test, or
>
> String can implement IArrayLike, but not string.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/closure-library/pull/655/files#r52824745>.
>